packages/solana/lib/utils.ts

```diff
@@ -0,0 +1,37 @@
+import { Block, Transaction, TxStatus } from '@liquality/types';
+import { BlockResponse, ParsedInstruction, ParsedTransactionWithMeta, RpcResponseAndContext, SignatureStatus } from '@solana/web3.js';
+
+export function parseBlockResponse(data: BlockResponse): Block {
+    return {
+        hash: data.blockhash,
+        timestamp: data.blockTime,
+        number: data.parentSlot + 1,
+        parentHash: data.previousBlockhash,
+        _raw: data,
+    };
+}
+
+export function parseTransactionResponse(
+    transaction: ParsedTransactionWithMeta,
+    signatureStatus?: RpcResponseAndContext<SignatureStatus>
+): Transaction {
+    const {
+        transaction: {
+            message: { instructions },
+            signatures,
+        },
+    } = transaction;
+
+    const [hash] = signatures;
+    const _instructions = instructions as ParsedInstruction[];
+
+    return {
+        hash,
+        value: _instructions?.[0]?.parsed?.info?.lamports || Number(_instructions?.[0]?.parsed?.info?.amount) || 0,
+        _raw: transaction,
+        ...(signatureStatus && {
+            confirmations: signatureStatus.value.confirmations,
+            status: signatureStatus.value.err ? TxStatus.Failed : TxStatus.Success,
```

Comment:
the `status` is a mandatory field and here if the signatureStatus is not defined neither will be the status.

probably if the `signatureStatus` is missing the status should be failed?

Reply:
The thing is there that `signatureStatus` always exists if we get `getTransactionByHash` however if we execute `getBlockByNumber` with `includeTx = true` only then is missing